### PR TITLE
torchdata 0.9.0 :snowflake:

### PR DESCRIPTION
**Pull request by psteyer**
torchdata 0.9.0 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-6039]
- dev_url:        https://github.com/pytorch/data/tree/v0.9.0
- conda_forge:    https://github.com/conda-forge/torchdata-feedstock
- pypi:           https://pypi.org/project/torchdata/0.9.0
- pypi inspector: https://inspector.pypi.io/project/torchdata/0.9.0

### Explanation of changes:

- new version number
- forcing pytorch 2.3.1 pinning due to `osx-64` pulling in an older version that is missing an import
- requires a newer portalocker: https://github.com/AnacondaRecipes/portalocker-feedstock/pull/4
- skipping some tests due to being resource intensive

[PKG-6039]: https://anaconda.atlassian.net/browse/PKG-6039?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 source:
   url: https://github.com/pytorch/data/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: c596db251c5e6550db3f00e4308ee7112585cca4d6a1c82a433478fd86693257
+  sha256: b547bbe848ad813cc5365fe0bb02051150bec6c7c4ee7bffd6b6d3d7bdeddd75
 
 build:
   number: 0
   # pytorch 2.0.x not available on ppc64le
-  skip: true  # [py<37 or ppc64le]
+  skip: true  # [py<39 or py>311 or s390x]
```
Comment (cbouss) [inline recipe/meta.yaml]: What is the reason for skipping `py>311`?
I see it could be portalocker. Could we build portalocker instead?

Comment (psteyer) [reply recipe/meta.yaml]: Yes, it looks like its due to:

```
portalocker-2.3.0-py38hca03da5_0 requires python >=3.8,<3.9.0a0, but none of the providers can be installed
```

Would building `portalocker latest (3.0.0)` be a better solution?

Comment (cbouss) [reply recipe/meta.yaml]: Yes

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
   version: {{ version }}
 
 source:
   url: https://github.com/pytorch/data/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: c596db251c5e6550db3f00e4308ee7112585cca4d6a1c82a433478fd86693257
+  sha256: b547bbe848ad813cc5365fe0bb02051150bec6c7c4ee7bffd6b6d3d7bdeddd75
 
 build:
   number: 0
   # pytorch 2.0.x not available on ppc64le
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -25,26 +25,25 @@ requirements:
...
+    - pytorch >=2.3.1
     - setuptools
     - wheel
   run:
     - python
     - requests
     - urllib3 >=1.25
     - portalocker >=2.0.0
-    - pytorch >=2.0.0
+    - pytorch >=2.3.1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: setup.py states about pytorch >=2, see https://github.com/pytorch/data/blob/v0.9.0/setup.py#L113
Where does `>=2.3.1` come from?

File: recipe/meta.yaml (hunk 2 above)
Comment (psteyer) [reply recipe/meta.yaml]: removing

File: recipe/meta.yaml (hunk 3 above)
Comment (psteyer) [reply recipe/meta.yaml]: I think this was left over from when I was testing  things, due to having build issues where it was pulling in older versions of `pytorch` for a specific version of python. I'll change it back

**Review by skupr-anaconda — APPROVED** [review]

**Review by msentissi — APPROVED** [review]

**Pull request merged**